tests/planner/ompl/test_OMPLSimplifier.cpp

```diff
@@ -0,0 +1,189 @@
+#include "OMPLTestHelpers.hpp"
+#include "../../constraint/MockConstraints.hpp"
+#include <aikido/planner/ompl/Planner.hpp>
+#include <aikido/planner/ompl/CRRT.hpp>
+#include <aikido/planner/ompl/CRRTConnect.hpp>
+#include <aikido/trajectory/Interpolated.hpp>
+#include <aikido/constraint/uniform/RnBoxConstraint.hpp>
+#include <aikido/constraint/CartesianProductSampleable.hpp>
+#include <aikido/constraint/CartesianProductTestable.hpp>
+#include <aikido/constraint/JointStateSpaceHelpers.hpp>
+#include <aikido/util/StepSequence.hpp>
+#include <aikido/planner/ompl/MotionValidator.hpp>
+#include <ompl/geometric/planners/rrt/RRTConnect.h>
+
+using StateSpace = aikido::statespace::dart::MetaSkeletonStateSpace;
+using aikido::planner::ompl::getSpaceInformation;
+using aikido::planner::ompl::CRRT;
+using aikido::planner::ompl::CRRTConnect;
+
+namespace {
+
+  template <typename T>
+  double computeTrajLength(T& traj,
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr stateSpace,
+      aikido::distance::DistanceMetricPtr& dmetric)
```

Comment:
`planOMPL` returns an `std::shared_ptr` while `simplifyOMPL` returns an `std::unique_ptr` `Interpolated` trajectory. Templated it to account for both types. since I will scratch planOMPL completely and make trajectories by hand I guess I can get rid of the templating.